### PR TITLE
Interactive map Issue #248

### PR DESCRIPTION
**Pull request by Gitek1**
This is a new pull request for panel map.  I added some functionality to current map panel, that after press on button named Request or even anywhere on the map, that map is resized over the rest of the content with different style and funcionality. This fullscreen map becomes interactive (google weather layer, earth integration on that google map, navigation controls, event's pop up for nodes, etc. ).

### CODE REVIEW

**Comment by mitar** [comment]
Interesting. But much more work should be done. :-)

First, I don't understand what is "earth integration on that google map"? What exactly does it provide?

Then, the full screen is nice, but I am not sure how useful is. You know, to steal whole screen just like that. So I would like to see it in two steps. First, original map panel should like it was, so just a map panel. Then there should be some way to enlarge it. And it should grow to overlay the content, like over everything, but that there is still for example 1cm around it. And then there in this large version can be some button for full screen.

You should show in some better way all layers which are available. So an open list of them on the side, for example.

**Comment by mitar** [comment]
Or simply don't do full screen for now. I am really not sure if this is useful (except that it is eye-candy).

Can you instead make an animation. Where map is enlarged to the big one version? And becomes interactive? And then when you minimize it, it goes back to panel map, not-interactive.

**Comment by mitar** [comment]
TODO.

**Comment by Gitek1** [comment]
Okey. I will do animaiton with JQUERY in map panel, so when we click the button, an overlay will be opened (whose contents will be loaded from other internal html page) over everything except over other panels. And that internal html  will contain that big interactive map)It will be ok?

**Comment by mitar** [comment]
No, just resize the map. Contents for the map we will load from specific API URL, where you will be able to get JSON data for various layers. Resize over the content, even panels.

**Comment by mitar** [comment]
Please write a comment when you update the pull request, because I don't get notification for that automatically.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.css [hunk 1]
```diff
@@ -1,4 +1,81 @@
-#map_canvas {
-    width: 450px;
-    height: 450px;
+#map_small {
+    width: 210px;
+    height: 200px;
+    display: inherit;
 }
+.centered {
+
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.css:7]: Be a bit more precise with style. So here is one unnecessary line. Ant there is not an empty line between blocks.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.css [hunk 2]
```diff
@@ -1,4 +1,81 @@
...
+    margin-right: auto; margin-left: auto;
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.css:8]: Also, this should go each into its own line.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.css [hunk 3]
```diff
@@ -1,4 +1,81 @@
...
+}
+#map_big {
+    background-color: #0000FF;
+    position: absolute;
+    top: 70px;
+    left: 70px;
+    right: 70px;
+    bottom: 70px;
+    display: none
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.css:17]: Semicolon at the end!

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.css [hunk 4]
```diff
@@ -1,4 +1,81 @@
...
+}
+
+#small_container.fullScreen{
+    width: 100%; height: 100%; text-align: center; margin: 50px 0px; padding: 0px;
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.css:21]: Too.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.css [hunk 5]
```diff
@@ -1,4 +1,81 @@
...
+}
+
+hr {
+    border: none;
+    border-top: 1px solid #e6e6e6;
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.css:26]: Do you use upper-case or lower-case written colors?

File: piplmesh/panels/map/templates/panel/map/panel.html [hunk 6]
```diff
@@ -3,22 +3,35 @@
...
-        </script>
-    {% endaddtoblock %}
-    <div id="map_info"></div>
-    <div id="map_canvas"></div>
-{% endblock %}
+{% addtoblock "js" %}<script src="http://maps.googleapis.com/maps/api/js?libraries=weather&sensor=false"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="https://www.google.com/jsapi" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="{% static "piplmesh/panel/map/googleearth.js" %}" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "css" %}<link href="{% static "piplmesh/panel/map/panel.css" %}" rel="stylesheet" type="text/css" media="screen" />{% endaddtoblock %}
+{% addtoblock "js" %}<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7/jquery.min.js"></script>{% endaddtoblock %}
```
Comment (mitar) [inline piplmesh/panels/map/templates/panel/map/panel.html:10]: We already have jQuery?

File: piplmesh/panels/map/templates/panel/map/panel.html [hunk 7]
```diff
@@ -3,22 +3,35 @@
...
-            /* ]]> */
-        </script>
-    {% endaddtoblock %}
-    <div id="map_info"></div>
-    <div id="map_canvas"></div>
-{% endblock %}
+{% addtoblock "js" %}<script src="http://maps.googleapis.com/maps/api/js?libraries=weather&sensor=false"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="https://www.google.com/jsapi" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="{% static "piplmesh/panel/map/googleearth.js" %}" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "css" %}<link href="{% static "piplmesh/panel/map/panel.css" %}" rel="stylesheet" type="text/css" media="screen" />{% endaddtoblock %}
```
Comment (mitar) [inline piplmesh/panels/map/templates/panel/map/panel.html:9]: Please list first CSS, then JS.

File: piplmesh/panels/map/templates/panel/map/panel.html [hunk 8]
```diff
@@ -3,22 +3,35 @@
...
+        'location': '{{ request.node.location|escapejs }}',
+        'url': '{{ request.node.url|escapejs }}'
+    };
+    /* ]]> */
+</script>
+{% endaddtoblock %}
+
+<div id="map_info"></div>
+<section id="small_container" class="small_container">
+    <div id="map_small" class = "centered"></div>
```
Comment (mitar) [inline piplmesh/panels/map/templates/panel/map/panel.html:29]: No spaces around `=`.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 9]
```diff
@@ -1,7 +1,137 @@
...
+var vl;
+var weather_visible = false;
+var infoWindow;
+var ge;
+google.load("earth", "1");
+var googleEarth;
+
+
+ document.onkeydown = function(evt) {
+ evt = evt || window.event;
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js:12]: Indent.

File: piplmesh/panels/map/static/piplmesh/panel/map/panel.js [hunk 10]
```diff
@@ -1,7 +1,137 @@
...
+ if (evt.keyCode == 27) {
+ toggleBigToSmalMap();
+ }
+ };
+
+
+function setWeatherVisible()
+{
```
Comment (mitar) [inline piplmesh/panels/map/static/piplmesh/panel/map/panel.js:20]: Put this on the line before. Please keep the same JavaScript code style as we are using elsewhere.

File: piplmesh/panels/map/templates/panel/map/panel.html [hunk 11]
```diff
@@ -3,22 +3,35 @@
...
-    <div id="map_info"></div>
-    <div id="map_canvas"></div>
-{% endblock %}
+{% addtoblock "js" %}<script src="http://maps.googleapis.com/maps/api/js?libraries=weather&sensor=false"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="https://www.google.com/jsapi" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="{% static "piplmesh/panel/map/googleearth.js" %}" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "css" %}<link href="{% static "piplmesh/panel/map/panel.css" %}" rel="stylesheet" type="text/css" media="screen" />{% endaddtoblock %}
+{% addtoblock "js" %}<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.7/jquery.min.js"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="{% static "piplmesh/panel/map/panel.js" %}" type="text/javascript"></script>{% endaddtoblock %}
+{% addtoblock "js" %}<script src="{% static "piplmesh/panel/map/jquery.fullscreen.js" %}" type="text/javascript"></script>{% endaddtoblock %}
```
Comment (mitar) [inline piplmesh/panels/map/templates/panel/map/panel.html:12]: This is not needed anymore?

**Comment by mitar** [comment]
It does not really work for me. It should be a modal panel overlaying everything. Also map is not really resized, it is just displayed in a corner of the panel.

For how to do overlay, you can check [this library](https://bitbucket.org/mitar/jquery-visage).

So code is messy and it is hard to read it, please format it nicely. Also things do not even work, so it is hard to check things. Are you sure you want to use Google Earth API? Why is this useful? It makes code here more complicated (I even didn't manage to see it in action), maybe this could be a separate pull request? So for now it would be enough work just to make this map "pop up" out of the webpage.

So, more work is needed here, definitely.

**Comment by mitar** [comment]
Oh, ugly ugly. You made two pull requests with same branch. I officially hate you.

**Comment by mitar** [comment]
(Think about using some of things from this pull requests, for example, weather layer, in new pull request.)

**Pull request merged**